### PR TITLE
Bugs/698 improve support and tests for date function

### PR DESCRIPTION
**Pull request by colewfrench**

### CODE REVIEW

File: EPPlus/FormulaParsing/Excel/Functions/DateTime/Date.cs [hunk 1]
```diff
@@ -22,25 +22,61 @@
...
-			date = date.AddDays((double)(day - 1));
-			return CreateResult(date.ToOADate(), DataType.Date);
+			var isValidYear = this.TryGetArgAsInt(arguments, 0, out int year);
+			var isValidMonth = this.TryGetArgAsInt(arguments, 1, out int month);
+			var isValidDay = this.TryGetArgAsInt(arguments, 2, out int day);
+			if (!isValidYear || !isValidMonth || !isValidDay)
+				return new CompileResult(eErrorType.Value);
+			if (year < 0 || year > 9999)
+				return new CompileResult(eErrorType.Num);
+			else if (year < 1900)
```
Comment (emdelaney) [inline EPPlus/FormulaParsing/Excel/Functions/DateTime/Date.cs]: Nice test coverage of the edge cases here (-1, 1899, 1900, etc).

File: EPPlusTest/FormulaParsing/Excel/Functions/DateTimeFunctions/DateTest.cs [hunk 2]
```diff
@@ -48,43 +48,606 @@ public void DateFunctionShouldReturnADate()
...
+		[TestMethod]
+		public void DateFunctionWithNullMonthAndNullDayReturnsCorrectResult()
+		{
+			var expectedDate = new DateTime(1999, 11, 30);
+			var func = new Date();
+			var args = FunctionsHelper.CreateArgs(2000, null, null);
+			var result = func.Execute(args, this.ParsingContext);
+			Assert.AreEqual(expectedDate.ToOADate(), result.Result);
+		}
+
```
Comment (emdelaney) [inline EPPlusTest/FormulaParsing/Excel/Functions/DateTimeFunctions/DateTest.cs]: You've got pretty good test coverage already so you don't have to add this, but I'd be curious to see a test case with obviously-insane input (say, DATE(2000, 25, 366), to ensure that (a) we add things in the correct order and (b) demonstrate that you can give the function things that aren't a literal day or month.
